crates/bevy_ecs/src/storage/resource.rs

```diff
@@ -25,7 +25,7 @@ pub struct ResourceData<const SEND: bool> {
 
 impl<const SEND: bool> Drop for ResourceData<SEND> {
     fn drop(&mut self) {
-        if self.is_present() {
+        if !SEND && self.is_present() {
```

Comment:
Comment here please explaining the logic.

Reply:
Added the comment.